numpy/ma/core.py

```diff
@@ -3121,6 +3121,13 @@ def __getitem__(self, indx):
             if isinstance(indx, basestring):
                 if self._fill_value is not None:
                     dout._fill_value = self._fill_value[indx]
+                    # If we're indexing a multidimensional field in a
+                    # structured array (such as dtype("(2)i2,(2)i1")),
+                    # indexing fill_value will yield a multi-element
```

Comment:
True, if the code repeats the access a lot and/or the field is a large array, it could cause some large overheads. So maybe you are right, but I am also not a big fan of silently throwing away info :/. Anyway, there are some people here who have a much better idea about what to do with masked arrays....